mainwindow.cpp

```diff
@@ -19,3 +54,529 @@ MainWindow::~MainWindow()
+
+    QString flightPathModifyNextId = ui->flightPathModifyNextIdEdit->text();
+    convertMessage(flightPathModifyNextId, MESSAGE_ID_PATH_MODIFY_NEXT_LD);
+
+    QString flightPathModifyPrevId = ui->flightPathModifyPrevIdEdit->text();
+    convertMessage(flightPathModifyPrevId, MESSAGE_ID_PATH_MODIFY_PREV_LD);
+
+    QString flightPathModifyId = ui->flightPathModifyIdEdit->text();
+    convertMessage(flightPathModifyId, MESSAGE_ID_PATH_MODIFY_LD);
+
+    QString homeBaseLatitude = ui->homeBaseLatitudeEdit->text();
+    QString homeBaseLongitude = ui->homeBaseLongitudeEdit->text();
+    QString homeBaseAltitude = ui->homeBaseAltitudeEdit->text();
+    QString homeBaseTurnRadius = ui->homeBaseTurnRadiusdit->text();
+    QString homeBaseWaypointType = enumSelection(ui->homeBaseWaypointTypeBox);
+
+    QList<QString> homeBaseInfo = {homeBaseLatitude,
+                                   homeBaseLongitude,
+                                   homeBaseAltitude,
+                                   homeBaseTurnRadius,
+                                   homeBaseWaypointType};
+    convertMessage(homeBaseInfo, MESSAGE_ID_HOMEBASE);
+
+
+    int numWaypoints = ui->setWaypointNumberEdit->value();
+    convertMessage(numWaypoints == 0 ? "" : QString::number(numWaypoints),
+                   MESSAGE_ID_NUM_WAYPOINTS);
+
+    if (!(ui->waypointScrollAreaWidgetContents->layout() == nullptr || numWaypoints == 0))
+    {
+        QFormLayout *layout = qobject_cast<QFormLayout *>(ui->waypointScrollAreaWidgetContents->layout());
+
+        for (int i{}; i < numWaypoints; i++)
+        {
+            QString waypointLatitude = qobject_cast<QLineEdit *>(layout->itemAt(i * 14 + 4)->widget())->text();
+            QString waypointLongitude = qobject_cast<QLineEdit *>(layout->itemAt(i * 14 + 6)->widget())->text();
+            QString waypointAltitude = qobject_cast<QLineEdit *>(layout->itemAt(i * 14 + 8)->widget())->text();
+            QString waypointTurnRadius = qobject_cast<QLineEdit *>(layout->itemAt(i * 14 + 10)->widget())->text();
+            QString waypointType = enumSelection(qobject_cast<QComboBox *>(layout->itemAt(i * 14 + 12)->widget()));
+
+            QList<QString> waypointInfo = {waypointLatitude,
+                                           waypointLongitude,
+                                           waypointAltitude,
+                                           waypointTurnRadius,
+                                           waypointType};
+
+            convertMessage(waypointInfo, MESSAGE_ID_WAYPOINTS);
+        }
+    }
+}
+
+
+void MainWindow::pigoFileChanged(const QString & path)
+{
+   if (allowReading){
+       if (QFile::exists(path)) {
+            watcher->addPath(path);
+        }
+
+      QFile file(path);
+      file.open(QFile::ReadOnly | QFile::Text);
+      QTextStream in(&file);
+      QString text = in.readAll();
+
+
+      std::string current_locale_text = text.toLocal8Bit().constData();
```

Comment:
Change it from `std::string current_locale_text = text.toLocal8Bit().constData();` to `std::string current_locale_text = text.toUtf8().constData()` when running the program on the Linux PC.